lib/db/CollisionDAO.js

```diff
@@ -80,6 +84,52 @@ WHERE collision_id = $(id)`;
     return { event, involved };
   }
 
+  /**
+   *
+   * @param {CentrelineType} centrelineType - type of centreline feature
+   * @param {number} centrelineId - ID of centreline feature
+   */
+  static async byCentreline(centrelineType, centrelineId) {
+    const params = { centrelineId, centrelineType };
+    const filters = [
+      'ec.centreline_id = $(centrelineId)',
+      'ec.centreline_type = $(centrelineType)',
+    ];
+    const sqlFilters = filters.join('\n  AND ');
+    const sqlEvents = `
+SELECT e.*
+FROM collisions.events e
+JOIN collisions.events_centreline ec ON e.collision_id = ec.collision_id
+WHERE ${sqlFilters}
+ORDER BY e.collision_id`;
+    const sqlInvolved = `
+SELECT i.*
+FROM collisions.involved i
+JOIN collisions.events_centreline ec ON i.collision_id = ec.collision_id
+WHERE ${sqlFilters}
+ORDER BY i.collision_id`;
+    const [rowsEvents, rowsInvolved] = await Promise.all([
+      db.manyOrNone(sqlEvents, params),
+      db.manyOrNone(sqlInvolved, params),
+    ]);
+    const events = rowsEvents.map(row => ({
+      ...normalizeEvent(row),
+      involved: [],
+    }));
+    let i = 0;
+    rowsInvolved.forEach((row) => {
+      const { collision_id: collisionId } = row;
+      while (events[i].collisionId < collisionId) {
+        i += 1;
+      }
+      if (events[i].collisionId === collisionId) {
+        const involved = normalizeInvolved(row);
+        events[i].involved.push(involved);
+      }
+    });
+    return events;
+  }
+
   static async byCentrelineSummary(
     centrelineId,
```

Comment:
for near-future work: we'll probably be getting rid of this method in favour of using `CollisionDAO.byCentreline` and performing necessary aggregations in JavaScript.

(we've used this strategy elsewhere - e.g. in Track Requests - to prioritize iteration velocity over super-production-quality performance so that we can get features out the door and in front of people quickly.)